src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -219,28 +299,53 @@ bool DirectoryService::VerifyPoWOrdering(const DequeOfShard& shards) {
     lastBlockHash =
         m_mediator.m_txBlockChain.GetLastBlock().GetBlockHash().asBytes();
   }
-  // Temporarily add the old ds to check ordering
-  m_allPoWs[m_mediator.m_DSCommittee->back().first] =
-      array<unsigned char, BLOCK_HASH_SIZE>();
 
-  vector<unsigned char> hashVec;
+  constexpr float MISORDER_TOLERENCE = 0.02f;  // 2 Percent tolerance
+  const uint32_t MAX_MISORDER_NODE =
+      std::ceil(m_allPoWs.size() * MISORDER_TOLERENCE);
```

Comment:
should u take some constant like the ds_committee size here instead of all pows size

Reply:
No, this is to verify the PoW submissions, not the ds comittee.

Reply:
yes but the tolerance level depends upon the PoW submissions recvd, should not it be a factor of the ds committee size?

Reply:
The ds committee size has no relationship with number of PoW submissions.

Reply:
I edited my comment, please see again

Reply:
From my test result,  the number of PoW submissions for DS leader and DS backup is the same, they are the same as total nodes in sharding structure. And only some received DS PoW, some didn't. And even the number of PoW is a little different, after multiply the factor of 0.02, the tolerance will be the same, unless the number of PoW difference over 50.